src/lib/SDK/sdk.js

```diff
@@ -232,4 +237,117 @@ export default class SDK {
     const signer = await this.getSigner();
     return signer.provider.getTransactionReceipt(txHash);
   }
+
+  /** Store data in ipfs
+   * @param {string} data data to store such as path to local file or url
+   * @returns {Promise<string>} Ipfs hash of the stored data
+   */
+  async storeFile(metadata) {
+    if (typeof metadata !== 'string' && metadata instanceof String === false) {
+      throw new Error(
+        errorLogger({
+          location: ERROR_LOG.location.SDK_store,
+          message: ERROR_LOG.message.data_must_be_string,
+        }),
+      );
+    }
+
+    if (!this.#ipfsClient) {
+      throw new Error(
+        errorLogger({
+          location: ERROR_LOG.location.SDK_store,
+          message: ERROR_LOG.message.invalid_ipfs_setup,
+        }),
+      );
+    }
+
+    return this.#ipfsClient.uploadFile({ source: metadata });
+  }
+
+  /** Store metadata in ipfs
+   * @param {string} data valid json metadata
+   * @returns {Promise<string>} Ipfs hash of the stored data
+   */
+  async storeMetadata(metadata) {
+    if (!isJson(metadata)) {
+      throw new Error(
+        errorLogger({
+          location: ERROR_LOG.location.SDK_store,
+          message: ERROR_LOG.message.data_must_be_valid_json,
+        }),
+      );
+    }
+
+    if (!this.#ipfsClient) {
+      throw new Error(
+        errorLogger({
+          location: ERROR_LOG.location.SDK_store,
+          message: ERROR_LOG.message.invalid_ipfs_setup,
+        }),
+      );
+    }
+    return this.#ipfsClient.uploadContent({ source: metadata });
+  }
+
+  /** Store data in ipfs
+   * @param {Array<any>} metadata an array of valid JSON Metadata
+   * @returns {Promise<string>} Ipfs hash of the stored data
+   */
+  async storeBaseURI(metadata) {
```

Comment:
I don't like this storeBaseURI it's confusing rather storeContractMetadata?

Reply:
`createFolder` might make more sense on what is actually being done